client/components/modal.jsx

```diff
@@ -0,0 +1,148 @@
+    }
+    this.setState({
+      addExercise: this.state.userInput,
+      userInput: ''
+    });
+  }
+
+  addSet() {
+    event.preventDefault();
+    this.setState(({
+      newSet: [...this.state.newSet, { reps: '', weight: '' }]
+    }));
+  }
+
+  removeSet(index) {
+    const newSet = this.state.newSet;
+    newSet.splice(index, 1);
+    this.setState({ newSet });
+  }
+
+  handleChange(index, event) {
+    const newSet = this.state.newSet;
+    newSet[index][event.target.name] = event.target.value;
+    this.setState({ newSet });
+  }
+
+  onType(event) {
+    const userInput = event.currentTarget.value;
+    const filteredExercises = this.state.exercises.filter(
+      exercise => exercise.exercise.toLowerCase().indexOf(userInput.toLowerCase()) > -1
+    );
+    this.setState({
+      activeSuggestions: 0,
+      filteredExercises,
+      showSuggestions: true,
+      userInput: event.target.value
+    });
+  }
+
+  onClick(event) {
+    this.setState({
+      activeSuggestions: 0,
+      filteredExercises: [],
+      showSuggestions: false,
+      userInput: event.currentTarget.innerText
+    });
+  }
+
+  render() {
+    if (this.state.addExercise.length > 0) {
+      return (
+          <div className="modal">
+            <div className="modal-content position-relative overflow-scroll">
+              <h1>Add Exercise</h1>
+              <Search state={this.state} onType={this.onType} click={this.onClick} addExercise={this.addExercise} />
+              <div className="row">
+                <form onSubmit={this.props.submitExercise}>
+                  <div className="row">
+                    <h3 className="margin-bottom-5">{this.state.addExercise}</h3>
+                  </div>
+                  {this.state.newSet.map((element, index) => (
+                  <div className="row justify-content-center align-items-flex-end" key={index}>
+                    <div className="column-third margin-right-10">
+                      <h5>Number of Reps (RM)</h5>
+                      <input className="input-width" type="text" name="reps" onChange={event => this.handleChange(index, event)}></input>
+                    </div>
+                    <div className="column-third">
+                      <h5>Weight (lbs)</h5>
+                      <input className="input-width" type="text" name="weight" onChange={event => this.handleChange(index, event)}></input>
+                      <div className="row justify-content-end margin-top-10">
+                        <div>
+                            <button type="button" onClick={() => this.addSet()}>Add Set</button>
+                        </div>
+                          {
+                            index
+                              ? <div className="margin-left-10">
+                                <button type="button" onClick={() => this.removeSet(index)}>Remove Set</button>
+                              </div>
+                              : null
+                          }
```

Comment:
Rather than use a ternary to choose between something and nothing, use `&&` to either include an element or not

```jsx
{ index > 0 &&
  <div> etc etc </div>
}
```

https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator